src/Elastic.Markdown/CrossLinks/CrossLinkResolver.cs

```diff
@@ -0,0 +1,93 @@
+// Licensed to Elasticsearch B.V under one or more agreements.
+// Elasticsearch B.V licenses this file to you under the Apache 2.0 License.
+// See the LICENSE file in the project root for more information
+
+using System.Collections.Frozen;
+using System.Diagnostics.CodeAnalysis;
+using System.Text.Json;
+using Elastic.Markdown.IO.Configuration;
+using Elastic.Markdown.IO.State;
+using Microsoft.Extensions.Logging;
+
+namespace Elastic.Markdown.CrossLinks;
+
+public interface ICrossLinkResolver
+{
+	Task FetchLinks();
+	bool TryResolve(Action<string> errorEmitter, Uri crossLinkUri, [NotNullWhen(true)] out Uri? resolvedUri);
+}
+
+public class CrossLinkResolver(ConfigurationFile configuration, ILoggerFactory logger) : ICrossLinkResolver
+{
+	private readonly string[] _links = configuration.CrossLinkRepositories;
+	private FrozenDictionary<string, LinkReference> _linkReferences = new Dictionary<string, LinkReference>().ToFrozenDictionary();
+	private readonly ILogger _logger = logger.CreateLogger(nameof(CrossLinkResolver));
+
+	public static LinkReference Deserialize(string json) =>
+		JsonSerializer.Deserialize(json, SourceGenerationContext.Default.LinkReference)!;
+
+	public async Task FetchLinks()
+	{
+		using var client = new HttpClient();
+		var dictionary = new Dictionary<string, LinkReference>();
+		foreach (var link in _links)
+		{
+			var url = $"https://elastic-docs-link-index.s3.us-east-2.amazonaws.com/elastic/{link}/main/links.json";
+			_logger.LogInformation($"Fetching {url}");
+			var json = await client.GetStringAsync(url);
```

Comment:
What happens if the url doesn't exit yet?

While we onboard repositories, not all of the might be available from the start.

Reply:
For now I'm only going to make `docs-content` mandatory to resolve. 

All others we'll blindly transform to urls for now. (will follow up with another PR to introduce this behavior).